### DISCUSSION TITLE
BroadcastChannel API for multi window support

### DISCUSSION THREAD

**Discussion by lucasyvas** [TanStack/query#1652] [Ideas] [answered]
Thanks for writing `react-query`! I was browsing through the discussions and noticed a particularly interesting suggestion to add IndexedDB as a storage option:

https://github.com/tannerlinsley/react-query/discussions/1638

My brain tingled a bit, because I've used RxDB before and love how it is able to synchronize changes between windows using the BroadcastChannel API. In the case of that project, they support multiple transports, which helps it work in other environments like Electron as well where one might want to use something other than BroadcastChannel. Prior art for reference:

https://github.com/pubkey/broadcast-channel

This may be a non-goal of the project, and that would be fine as well! Just thought it could be super powerful in multi-window setups.

**Comment by jonathanstanley** [comment, marked as answer]
see https://github.com/tannerlinsley/react-query/pull/1793
